primer/primitives
# Add path aliases

## Summary

I added path aliases to avoid using relative paths like `../../src/test-utilities/` and make it easier to experiment with different directory structures.

```js
"~types/*": ["types/*"],
"~build-config/*": ["config/*"],
"~test-utilities/*": ["src/test-utilities/*"]
```

## List of notable changes:

- **added** path aliases to `tsconfig.json`
- **updated** build script to register paths
- **updated** jest config to use paths
- **updated** paths in all files


## What should reviewers focus on?

- changes make sense
- any idea why this config https://github.com/primer/primitives/blob/53f9e15d85ea4f76fd0e5997b43755d48e7da733/tsconfig.json for `ts-node` to use aliases does not work?
It is recommended: https://typestrong.org/ts-node/docs/paths/ but it only works for me with `ts-node -r tsconfig-paths/register` 😢 

## Steps to test:

1. Clone branch
1. `yarn`
1. check out files or run `npm t` or `npm run build:color-tokens`

## Supporting resources (related issues, external links, etc):
- https://typestrong.org/ts-node/docs/paths/

## Contributor checklist:


## Reviewer checklist:
